www/src/components/ReactPlayground.js

```diff
@@ -114,172 +117,168 @@ const StyledExample = styled('div')`
 
 let uid = 0;
 
-class Editor extends React.Component {
-  state = { ignoreTab: false };
-
-  id = `described-by-${++uid}`;
-
-  handleKeyDown = event => {
-    const { key } = event;
-
-    if (this.state.ignoreTab && key !== 'Tab' && key !== 'Shift') {
-      if (key === 'Enter') event.preventDefault();
-      this.setState({ ignoreTab: false });
-    }
-    if (!this.state.ignoreTab && key === 'Escape') {
-      this.setState({ ignoreTab: true });
-    }
-  };
-
-  handleFocus = e => {
-    if (e.target !== e.currentTarget) return;
-    this.setState({
-      ignoreTab: !this.mouseDown,
-      keyboardFocused: !this.mouseDown,
-    });
-  };
-
-  handleMouseDown = () => {
-    this.mouseDown = true;
+function Editor() {
+  const [focused, setFocused] = useState(false);
```

Comment:
So we just track this in-JS rather than with CSS?

Alternatively we can change the above to `:focus-within`.